core: write down some stdcm debugging tips (#11305)
OpenRailAssociation/osrd: core/stdcm_debugging_tips.md

2. In prod environments: find a trace_id using datadog ([private link for sncf envs](https://claps.datadoghq.eu/apm/traces?query=service%3Aosrd-editoast%20-status%3Aok&agg_m=count&agg_m_source=base&agg_t=count&cols=core_service%2Ccore_resource_name%2Clog_duration%2Clog_http.method%2Clog_http.status_code&fromUser=false&historicalData=false&messageDisplay=inline&query_translation_version=v0&sort=desc&spanType=all&storage=hot&view=spans&start=1743426877565&end=1743427777565&paused=false)).
Then use the `/api/stdcm_log?trace_id=id` endpoint. The json content needs to be trimmed to only keep the request payload.

## Reproduce request payloads

`java -jar build/libs/osrd-all.jar reproduce-request --stdcm-payload-path stdcm_payload.json --railjson france.json`

Inputting the infra is optional, if not specified we fetch the infra from a locally running editoast. But the infra_id needs to be correct in the request payload.

## Limit computation time

The pathfinding step can take *a while*. It can help a lot to reduce the numbers of possibilities while still reproducing the bug.

> One way to do this is to exclude any path that diverges from the bugged case. If the solution takes the path `a -> b -> C`,

Reviewer: ```suggestion
One way to do this is to exclude any path that diverges from the bugged case. If the solution takes the path `a -> b -> c`,
```

> we can immediatly exclude paths that start from `a -> x`. This generally speeds up the pathfinding step to be near instant.

Reviewer: ```suggestion
we can immediatly exclude paths that start from `a -> x`. This generally speeds up the pathfinding step to be almost instantaneous.
```

It's important to check that the bug is still reproduced though, it's not always the case.

I generally do this by logging blocks IDs used in the solution at the post-processing step (including lookahead), then inside `InfraExplorer.extend()`,
I add a `return false` if any block isn't in that list. I edit the code locally for both of these changes.

This also helps with breakpoints during the exploration, as we're less likely to break on irrelevant paths.

It can also be possible to filter using the times used, but in my experience this often changes the result.

## The case of `mismatch between exploration and postprocessing`

This error is raised when assumptions made during the explorations are proven to be false during the post-processing.

> More specifically, when thought we've found an opening for a solution, but we couldn't converge to a

Reviewer: ```suggestion
More specifically, when we thought we'd found an opening for a solution, but we couldn't converge to a
```

full simulation that's free of conflict.

This can have two causes:

1. The engineering allowances aren't possible
2. Incremental conflict detection / spacing requirement generation gives different results during exploration and post-processing (or we don't handle them properly)

Option 1 is easy to identify as it will be logged as a warning before the error (sometimes it works out fine). 

Option 2 is more annoying. When it happens, there's no way around it: we need to take a notebook and write everything down to identify what differs.
Using conditional breakpoints at post-processing can tell us which zone ID is conflicting and when. Other conditional breakpoints

> call tell us what happens during the exploration. 

Reviewer: ```suggestion
can tell us what happens during the exploration. 
```
